File: v13/ATLAS/scripts/test_vector.ts
import { Ascon } from 'ascon-js';

function bytesToHex(bytes: Uint8Array): string {
    return Array.from(bytes).map(b => b.toString(16).padStart(2, '0')).join('');
}

async function main() {
    const key = new Uint8Array(16);
    const nonce = new Uint8Array(16);
    const ad = new Uint8Array(0);
    const plaintext = new Uint8Array(0);

    const c = await Ascon.encrypt(key, nonce, plaintext, ad);
    console.log(`TS_Zero: ${bytesToHex(c)}`);
}

main();

File: v13/ATLAS/scripts/test_vector.py
import ascon
import sys


def main():
    key = bytes([0] * 16)
    nonce = bytes([0] * 16)
    ad = b""
    plaintext = b""

    # Ascon-128
    c = ascon.encrypt(key, nonce, ad, plaintext, variant="Ascon-128")
    print(f"Py_128_Zero: {c.hex()}")

    # Ascon-128a
    c_a = ascon.encrypt(key, nonce, ad, plaintext, variant="Ascon-128a")
    print(f"Py_128a_Zero: {c_a.hex()}")


if __name__ == "__main__":
    main()
